src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -13,6 +13,55 @@ public enum JsonCommentHandling : byte
         Disallow = (byte)0,
         Skip = (byte)2,
     }
+    public sealed partial class JsonDocument : System.IDisposable
+    {
+        internal JsonDocument() { }
+        public System.Text.Json.JsonElement RootElement { get { throw null; } }
+        public void Dispose() { }
+        public static System.Text.Json.JsonDocument Parse(System.ReadOnlyMemory<byte> utf8Json, System.Text.Json.JsonReaderOptions readerOptions) { throw null; }
+    }
+    public partial struct JsonElement
+    {
+        private object _dummy;
+        public System.Text.Json.JsonElement this[int index] { get { throw null; } }
+        public System.Text.Json.JsonElement this[System.ReadOnlySpan<byte> utf8PropertyName] { get { throw null; } }
+        public System.Text.Json.JsonElement this[System.ReadOnlySpan<char> propertyName] { get { throw null; } }
+        public System.Text.Json.JsonElement this[string propertyName] { get { throw null; } }
+        public System.Text.Json.JsonTokenType Type { get { throw null; } }
+        public System.Text.Json.JsonElement.ChildEnumerator EnumerateChildren() { throw null; }
+        public int GetArrayLength() { throw null; }
+        public bool GetBoolean() { throw null; }
+        public double GetDouble() { throw null; }
+        public int GetInt32() { throw null; }
+        public long GetInt64() { throw null; }
+        public string GetString() { throw null; }
+        [System.CLSCompliantAttribute(false)]
+        public ulong GetUInt64() { throw null; }
+        public static explicit operator bool (System.Text.Json.JsonElement element) { throw null; }
+        public static explicit operator double (System.Text.Json.JsonElement element) { throw null; }
+        public static explicit operator int (System.Text.Json.JsonElement element) { throw null; }
+        public static explicit operator long (System.Text.Json.JsonElement element) { throw null; }
+        public static explicit operator string (System.Text.Json.JsonElement element) { throw null; }
+        [System.CLSCompliantAttribute(false)]
+        public static explicit operator ulong (System.Text.Json.JsonElement element) { throw null; }
+        public override string ToString() { throw null; }
+        public bool TryGetProperty(System.ReadOnlySpan<byte> utf8PropertyName, out System.Text.Json.JsonElement value) { throw null; }
+        public bool TryGetProperty(System.ReadOnlySpan<char> propertyName, out System.Text.Json.JsonElement value) { throw null; }
+        public bool TryGetProperty(string propertyName, out System.Text.Json.JsonElement value) { throw null; }
+        public bool TryGetRawData<T>(out System.ReadOnlyMemory<T> rawData) where T : struct { throw null; }
```

Comment:
Why is this generic?  Are we imagining a future where JsonDocument also Parses `ReadOnlyMemory<char>`?  Anything beyond that?  Seems like maybe this should just be TryGetRawData working on `ReadOnlyMemory<byte>`, and if in the future we add Parse'ing support for `ReadOnlyMemory<char>`, we could add another overload.

Reply:
It's my current working thought on future expansion to char/Rune, yes.  The problem is your next question, actually... BigInteger/the-mythical-BigRational; making it so that anyone who writes an extension method has some form of "maybe there's other work you should be doing" and isn't guaranteed broken in the next version.

I apparently neglected to add TryCopyRawData, which is the "if you can't get the memory, I'll transcode it for you" fallback